lib/internal/loader/ModuleRequest.js

```diff
@@ -45,13 +46,41 @@ loaders.set('cjs', async (url) => {
 // through normal resolution
 loaders.set('builtin', async (url) => {
   const ctx = createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading BuiltinModule ${url.pathname}`);
-    const exports = NativeModule.require(url.pathname);
+    debug(`Loading BuiltinModule ${url}`);
+    const exports = NativeModule.require(url.substr(5));
     reflect.exports.default.set(exports);
   });
   return ctx.module;
 });
 
+loaders.set('addon', async (url) => {
+  const ctx = createDynamicModule(['default'], url, (reflect) => {
+    debug(`Loading NativeModule ${url}`);
+    const module = { exports: {} };
+    const pathname = internalURLModule.getPathFromURL(new URL(url));
+    process.dlopen(module, _makeLong(pathname));
+    reflect.exports.default.set(module.exports);
+  });
+  return ctx.module;
+});
+
+loaders.set('json', async (url) => {
+  const ctx = createDynamicModule(['default'], url, (reflect) => {
+    debug(`Loading JSONModule ${url}`);
+    const urlObj = new URL(url);
+    const pathname = internalURLModule.getPathFromURL(urlObj);
+    const content = fs.readFileSync(pathname, 'utf8');
```

Comment:
Style nit: in this case you assign to `urlObj` but in the `.set("cjs")` case you invoke as an argument to `getPathFromURL`.

Reply:
Thanks for review. At one point this was to reuse it, but that changed, certainly can make this consistent.